src/metabase/models/setting.clj

```diff
@@ -1342,3 +1343,42 @@
            (map (fn [[setting-name]]
                   [setting-name (get setting-name)])))
      @registered-settings)))
+
+(defn- set-via-env? [setting]
+  (contains? env/env (setting-env-map-name setting)))
```

Comment:
This will (unfortunately?) pick up `.lein-env` overrides as well. 

Unfortunately `System/getenv` can't get updated by `with-temp-env-var-value`, so we have to check the environ var directly. We *could* check against `(env/read-env-file ".lein-env")`, but that seems to over complicate things.

Since the file is "not officially supported" I think this is fine?

Reply:
I think the implementation of this function is coupled to how we get environment variables and when we choose to use them, when you could use a higher level function `env-var-value` that hides these details. This also contains a very subtle bug, because there are settings that are not "site-wide", so even if they are in the environment we would not use them. `env-var-value` contains the logic for excluding non-site-wide settings, so using it will fix the bug.
```suggestion
  (some? (set-via-env-var? setting)
```

There is actually already an instance where a function like this is defined:
https://github.com/metabase/metabase/blob/b34970a2c27cb78ab09996475fab1d0a56b66a02/src/metabase/models/setting.clj#L1240
Side-note: I think this contains another bug: `boolean` should be changed to `some?`

You should also change this function to use your new `set-via-env?` function so we don't have duplicated logic.